### PR TITLE
task10 Code for reduce only

### PR DESCRIPTION
**Pull request by 443n0511**
### inputの値をoutputの値として出力すること

input
```
[
  { id: "a", value: 1 },
  { id: "b", value: 2 },
  { id: "c", value: 3 }
]
```
output
```
{
  ids: ["a", "b", "c"],
  entities: {
    a: { value: 1 },
    b: { value: 2 },
    c: { value: 3 }
  }
}
```
#### [stackblitz](https://stackblitz.com/edit/task10-ekysq7?file=users%2Fkimurayoshimi%2F10_2.js)  
5ceefa8

### Run-time commands
```
node ./users/kimurayoshimi/10.js
```

### CODE REVIEW

File: users/kimurayoshimi/10_2.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const a = (b) =>
+  b.reduce((target, key) => {
+    target["ids"] = b.map((value) => value.id);
```
Comment (kenmori) [inline users/kimurayoshimi/10_2.js:3]: この中でmap使ってます。これ使わないできる認識です

Comment (443n0511) [reply users/kimurayoshimi/10_2.js]: I thought about it, but I couldn't figure it out. 
I'll report the code halfway through.
77fde13

File: users/kimurayoshimi/10_2.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const a = (b) =>
-  b.reduce((target, key) => {
-    target.ids = [target.ids] + [key.id];
+  b.reduce((target, key, index) => {
+    target.ids = [...([target.ids] + key.id)];
```
Comment (443n0511) [inline users/kimurayoshimi/10_2.js:3]: output 
```bash
{
  ids: [ 'a', ',', 'b', 'c' ],
  entities: { a: { value: 1 }, b: { value: 2 }, c: { value: 3 } }
}
```
There is an extra `,` in it.

File: users/kimurayoshimi/10_2.js [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+    (target, key, index) => {
+      target.ids[index] = key.id;
+      target.entities = {
+        ...target.entities,
+        [key.id]: { value: key.value },
+      };
+      return target;
+    },
+    { ids: [] }
+  );
```
Comment (443n0511) [inline users/kimurayoshimi/10_2.js:12]: I changed the code to pass an empty array as the initial value and put the key.id in the [index] of the array.
```js
   target.ids[index] = key.id;
```

**Comment by kenmori** [comment]
👍

**Pull request merged**